packages/orm/src/relations/ManyToOneReference.ts

```diff
@@ -116,7 +116,8 @@ export class ManyToOneReferenceImpl<T extends Entity, U extends Entity, N extend
   }
 
   get isLoaded(): boolean {
-    return this._isLoaded;
+    // Even if `.load()` has not been called, `.get` will detect
+    return this._isLoaded || this.current() === undefined || !!this.maybeFindEntity();
```

Comment:
I'm kind of surprised it didn't already do this, but if we called `m2o.get` on a technically-unloaded m2o, and it would the entity in the EM anyway, it would go ahead and return it.

We were already relying on this as basically the way to preload m2os (which is the mechanism the recursive dataloaders use), but the `isLoaded` wasn't updated to handle the "I didn't have `.load()` called but turns out I am actually loaded" scenario.